finalproject_v1.3.py

```diff
@@ -2,140 +2,213 @@
+            startSession(num+1, selectSession) #Calls startExperiment func
+            
+    visual.TextStim(win=mywin, text='End Of Experiment', pos=[0,0], wrapWidth=500).draw()
+    mywin.flip()
+    core.wait(2.0)
+    core.quit()
+    
 # converts word bank into list of individual words
-with open('wordBank.txt') as f:
-    wordCue = [word for line in f for word in line.split()]
+#with open('wordBank.txt') as f:
+#    wordCue = [word for line in f for word in line.split()]
 
 
 # Create all basic objects (window, fixation-cross, timer)
-timer = core.Clock()
-mywin = visual.Window(fullscr = True, monitor="testMonitor", units="deg")
-fixCross = visual.TextStim(mywin, text = '+', height = 0.1)
+
+#fixCross = visual.TextStim(mywin, text = '+', height = 0.1)
 
 # Key list variables that participants will use to indicate whether word is related to self or not
 # right = self, left = other, q = quit game; the key left and right refer to arrows keys
-keyL = ["right", "left", "q"]
+# keyL = ["right", "left", "q"]
 
 # Other global variables
-trial = 0
+# trial = 0
 
 
-def startExperiment():
+def startSession(sessionNum, selectSession):
 
     '''
-    This is the first function to be called.
+    This function to is called from the main function.
     
-    Part 1: It will display instructions for the experiment.
-
-    Part 2: Initializes word list for all sessions
+    Initalizes wordlist as values from wordBank.txt
+    
+    Part 1: It will display instructions for each session, the instruction is selected 
+            from instructions list based on selectSession value. 
 
-    Part 3: Calls practice func for each of the sessions
+    Part 2: Initializes word list for each session, wordlist is selected using selectSession
+            PS: SELF and OTHER has two extra characters to make it uniform with other
+                words from the wordBank
 
-    Part 4: Indicates end of experiment, waits 5 secs and closes the application
+    Part 3: Wordlist is shuffled and practice func is called with 
+            sessionNum,wordlist,isPractice,selectSession in [4,5] : reversed
+            Practice is called again for sets 3 and 5 with isPractice as False 
+            (this time the data is recorded)
     '''
 
+    wordlist = open("wordBank.txt").read().split()
     # Part 1
-    instructions = ['Welcome to our experiment you will see a series of words \nwhere you will press `->` for if you think the words pertains to yourself \nand `<-` if it does not \npress any key to continue']
-    pos = [[-9,8],[-4,6],[-4,4],[-6,2]]
-    for i in range(len(instructions)):
-        visual.TextStim(win=mywin, text=instructions[i], pos=[i], wrapWidth=500).draw()
+    instructions = [
+        'Welcome to our experiment you will see a series of words \nwhere you will press `p` if the word is "SELF"\nand `q` if the word is "OTHER" \npress any key to continue\n\n\nPlease make sure to press the right key',
```

Comment:
instead of positive/ negative , can go with pleasant and unpleasant for the instructions